frisc/src/extras/extras.fil

```diff
@@ -0,0 +1,40 @@
+// extra components to make things simpler to write
+extern "extras.sv" {
+  comp onehotmux[#WIDTH]<G:L-(G),?L:1=G+1>(
+    @[G,L] sel:8,
+    @[G,L] in0: #WIDTH,
+    @[G,L] in1: #WIDTH,
+    @[G,L] in2: #WIDTH,
+    @[G,L] in3: #WIDTH,
+    @[G,L] in4: #WIDTH,
+    @[G,L] in5: #WIDTH,
+    @[G,L] in6: #WIDTH,
+    @[G,L] in7: #WIDTH
+  ) -> (
+    @[G,L] out: #WIDTH
+  ) where L > G, #WIDTH > 0;
+
+  comp rightshifter[#WIDTH]<G:L-(G),?L:1=G+1>(
+    @[G,L] in: #WIDTH,
+    @[G,L] shamt: 5,
+    @[G,L] isArith:1
+  ) -> (
+    @[G,L] out:#WIDTH
+  ) where L > G, #WIDTH > 0;
+
+  comp addsub[#WIDTH]<G:L-(G),?L:1=G+1>(
+    @[G,L] inA: #WIDTH,
+    @[G,L] inB: #WIDTH,
+    @[G,L] isAdd:1, // 1 if add, 0 if sub
+  ) -> (
+    @[G,L] out: #WIDTH
+  ) where L > G, #WIDTH > 0;
+
+  comp Ternary[#WIDTH]<G:L-(G),?L:1=G+1>(
+    @[G,L] guard:1,
+    @[G,L] opTrue: #WIDTH,
+    @[G,L] opFalse: #WIDTH
+  ) -> (
+    @[G,L] out: #WIDTH
+  ) where L > G, #WIDTH > 0;
+}
```

Comment:
The last three seems like they should be implementable within Filament itself. Any reason to implement them in Verilog? Also, might be worth making the `onehotmux` a parameterized primitive one day (not necessarily in this PR)

Reply:
For the shifter, I wanted to use Verilog's right arithmetic shift operator (`>>>`). For the other two, I didn't have a specific reason I implemented them in Verilog, so I'll rewrite them in Filament.

Reply:
Good point! We should have an arithmetic shift operation in the standard library as well!